test/Transfers.test.ts

```diff
@@ -0,0 +1,69 @@
+import { ethers } from "hardhat";
+import { expect } from "chai";
+
+describe("Transfer", function () {
+    let owner: any;
+    let addr1: any;
+    let TestTransfers: any;
+    let testTransfersToken: any;
+    let WETH: any;
+    let TestERC20: any;
+    let tokenWeth: any;
+    let tokenA: any;
+
+    beforeEach(async function () {
+        [owner, addr1] = await ethers.getSigners();
+        TestERC20 = await ethers.getContractFactory("TestERC20");
+        tokenA = await TestERC20.deploy("Token A", "TOKA");
+
+        WETH = await ethers.getContractFactory("WETH9");
+        tokenWeth = await WETH.deploy();
+
+        await tokenWeth.deposit({value: ethers.utils.parseEther('100')});
+
+        TestTransfers = await ethers.getContractFactory("TestTransfers");
+        testTransfersToken = await TestTransfers.deploy(tokenWeth.address);
+
+        await tokenWeth.transfer(testTransfersToken.address, 5000);
+        await tokenA.transfer(testTransfersToken.address, 5000);
+
+        await testTransfersToken.deployed();
+        await tokenWeth.deployed();
+        await tokenA.deployed();
+    })
+
+    it("#unwrapWETH", async function () {
+        let currentAddr1Balance = await ethers.provider.getBalance(addr1.address)
+
+        expect((await tokenWeth.balanceOf(testTransfersToken.address)).toNumber()).to.equal(5000);
+        expect(await testTransfersToken.testUnwrapWETH(1000, addr1.address)).to.be.ok;
+        expect((await tokenWeth.balanceOf(testTransfersToken.address)).toNumber()).to.equal(0);
+        expect(await ethers.provider.getBalance(addr1.address)).to.equal(currentAddr1Balance.add(ethers.BigNumber.from(5000)));
+    })
+
+    it("#refundETH", async function () {
+        let prevBalance = await ethers.provider.getBalance(owner.address);
+        await testTransfersToken.testUnwrapWETH(1000, testTransfersToken.address);
+        expect(await testTransfersToken.testRefundETH()).to.be.ok;
+        expect(await ethers.provider.getBalance(owner.address)).to.not.equal(prevBalance);
```

Comment:
test that the new balance matches by the expected increased amount

Reply:
Balance of owner is getting populated in between the transaction, therefor not able to determine the exact balance of the owner.

Reply:
let prevBalanceOwner = await ethers.provider.getBalance(owner.address);
await testTransfersToken.testUnwrapWETH(1000, testTransfersToken.address);
let prevBalanceTestTransfersToken = await ethers.provider.getBalance(testTransfersToken.address);
expect(await testTransfersToken.testRefundETH()).to.be.ok;
expect(await ethers.provider.getBalance(owner.address)).equal(prevBalanceOwner + prevBalanceTestTransfersToken);